### ISSUE TITLE
integrating other famous components

### ISSUE THREAD

**Issue by dcsan** [raix/Meteor-famono#100]
what would be involved to get famono to allow use of the various new famo.us addons people are developing? for example FlexScrollView.js

https://github.com/IjzerenHein/famous-flex/blob/master/src/FlexScrollView.js

```
 * Inherited from: [ScrollController](./ScrollController.md)
 * @module
 */
define(function(require, exports, module) {
```

because of the requires its not a simple Drop in and use thing...

**Comment by raix** [comment]
It should be possible, something like:

Add in the `lib/smart.require`
```js
"Flex": {
  "git": "https://github.com/IjzerenHein/famous-flex.git",
  "root": "src"
}
```

```js
  var FlexScrollView = require('Flex/FlexScrollView');
```

Its completely untested - but this pattern should work

**Comment by gadicc** [comment]
Yeah, although I haven't tested the above I regularly use Famono to bring in other Famo.us modules and it works great :)  The only reason we don't use that pattern in fview-* plugins is to support non-Famono users.

Also, in case you intend to use with famous-views, see also:
* https://github.com/IjzerenHein/famous-flex/issues/44
* https://github.com/gadicc/meteor-famous-views/issues/178

**Comment by raix** [comment]
Kudos to @IjzerenHein - looks alot better in performance, seems like a bit of a rewrite?

We've been close to use famous in a project but the scroll failed in performance + and api's break about April 2015 - but I really hope they get there. (we were told by one of their team members)

**Comment by raix** [comment]
Btw. at some point we should perhaps create a tool to convert famous into tiny packages - rigging dependencies etc. / converting it into the Meteor package system - this would beat famono and other famous packages I think...

**Comment by gadicc** [comment]
@raix, yeah I'm also worried about the transition to mixed mode :>  and although i haven't tested it personally, famous on android < 4 should be fine with crosswalk, although obviously the apk will be a lot bigger.

is lots of tiny packages really better than famono?  what if only one small part of code from each package is used?  or you mean super tiny like one package for each file?  i guess i can see what you mean but from a dev experience and impact perspective, i really think famono is magical :)

**Comment by raix** [comment]
running chrome on android doesn't help performance only consistency.

yep, pr. file :) - famono is ok, but it would be nice to be able to use the meteor constraint resolver.

**Comment by jordangarside** [comment]
On Performance, @raix 
Interestingly performance is still terrible on my Z3 when I compile (20-30fps with simple effects NavDrawer/Scrollviews etc.). BUT, when I run the project straight from chrome on my phone without compiling, I get a nice 60fps and it feels like a native application. There is something to be desired when new surfaces are created (pagination slows down to ~35fps in chrome when loading new surfaces).
I still have yet to see if Crosswalk has the same effect as running in the chrome browser, but it is strange because I'm on Android 4.4 and my user agent string in the compiled version reports it's using Chrome.

On Flex, @gadicc
Have you heard any word about progress on someone integrating it? I was using Scrollview and ran into the problems, then I had major issues with infiniteScrollView, so I need to use Flex. At the moment it looks like I'm going to have to just render much of my content from Javascript instead of using helpers if I want to use Flex.

**Comment by gadicc** [comment]
@jordangarside, yeah, since Android 4.4 KitKat, the [default WebView](https://developer.chrome.com/multidevice/webview/overview) is based on Chromium rather than the old WebKit implementation.  So a lot of people consider Crosswalk to be more useful for Android < 4.4, although Crosswalk does offer an extended API (for hardware, etc), and as I just noticed in that link, the Chromium WebView doesn't have full feature parity with like-numbered Chrome releases.  Note that Android L's webview will have WebGL :)

Re flex, there are no updates beyond what's on that earlier thread unfortunately.  I'm going to look at it again after I get through my backlog from the holidays.  I still suspect it won't be too hard to wrap if you only support one of the modes, but I can't look into yet unfortunately :(

**Comment by gadicc** [comment]
@raix, re performance (besides just consistency), isn't the JS engine (in Chrome vs old webkit) much faster?  I've never bothered to look for benchmarks, always just kind of assumed that.

**Comment by raix** [comment]
@gadicc on my old android 4.0.4 the builtin webkit seems to run a bit faster than chrome - but I cant fire up any timeline comparison (one of the reasons chrome is installed)
but there was a visual difference in small frame drops.

For me consistency is the main issue - android made me cry so many times with weird translate glitches on specific distributions (not versions)

We get performance by limiting dom read/writes/manipulation and use composite layers with care.

**Comment by gadicc** [comment]
@raix, looking forward to the day when we look back and laugh about how hard it used to be to write great apps that worked consistently on all devices.  hope it's in our lifetime :)

@dcsan, @jordangarside - re famous-flex, just announced an early release in https://github.com/gadicc/meteor-famous-views/issues/178#issuecomment-73072352.  only in mjn:famono right now for prototyping... but now that the basics are up i'll work out the best way to integrate with raix:famono, i expect even tomorrow (for a working but less-than-ideal implementation :)).

**Comment by raix** [comment]
@gadicc let it be a sunny day by the ocean with a cold drink - I'm hoping 2 years from now

**Comment by gadicc** [comment]
@raix and on our yachts :)  i could wait 2 years for that :>

everyone else -- v0.0.2 is out with famono support.
